File: mobile-client/ios/App/Catalog/CatalogNavigationBridge.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
// Catalog-level navigation owner.
public struct CatalogNavigationBridge: Sendable {
    public let feature: CatalogFeatureBridge
    public let routeMap: CatalogCommerceRouteMap
    public let vendor: VendorNavigationBridge
    public let product: VendorProductBridge

    public init(
        feature: CatalogFeatureBridge,
        routeMap: CatalogCommerceRouteMap,
        vendor: VendorNavigationBridge,
        product: VendorProductBridge
    ) {
        self.feature = feature
        self.routeMap = routeMap
        self.vendor = vendor
        self.product = product
    }
}

File: mobile-client/android/app/src/main/java/com/smartresponsor/mobile/app/catalog/CatalogNavigationBridge.kt
package com.smartresponsor.mobile.app.catalog

import com.smartresponsor.mobile.app.vendor.VendorNavigationBridge
import com.smartresponsor.mobile.app.vendor.VendorProductBridge

/**
 * Marketing America Corp. Oleksandr Tishchenko
 *
 * Catalog-level navigation owner.
 *
 * Catalog remains a primary Dashboard section while routing commercial
 * discovery into Vendor-owned commerce flows.
 */
class CatalogNavigationBridge(
    private val feature: CatalogFeatureBridge,
    private val routeMap: CatalogCommerceRouteMap,
    private val vendor: VendorNavigationBridge,
    private val product: VendorProductBridge,
) {
    fun feature(): CatalogFeatureBridge = feature

    fun routeMap(): CatalogCommerceRouteMap = routeMap

    fun vendor(): VendorNavigationBridge = vendor

    fun product(): VendorProductBridge = product
}
